test/broadcasting.jl

```diff
@@ -0,0 +1,364 @@
+module TestBroadcasting
+
+using Test, DataFrames
+
+refdf = DataFrame(ones(3, 5))
+
+@testset "normal data frame and data frame row in broadcasted assignment - one column" begin
+    df = copy(refdf)
+    df[1] .+= 1
+    @test df.x1 == [2, 2, 2]
+    @test df[2:end] == refdf[2:end]
+
+    dfv = @view df[1:2, 2:end]
+    dfv[1] .+= 1
+    @test dfv.x2 == [2, 2]
+    @test all(Matrix(dfv[2:end]) .== ones(size(dfv[2:end])...))
```

Comment:
There are still lots of `all` calls which I think can be avoided.

Reply:
OK - I will look through them.

Reply:
I have cleaned out all `all` that were possible to remove.